_std/defines/component_defines/component_defines_datum.dm

```diff
@@ -28,6 +28,10 @@
 	#define COMSIG_AREA_ACTIVATED "area_activated"
 	/// area's active var set to false (when all clients leave)
 	#define COMSIG_AREA_DEACTIVATED "area_deactivated"
+	/// whenever a mob enters an area
+	#define COMSIG_MOB_ENTERED_AREA "mob_entered_area"
+	/// whenever a mob exits an area
+	#define COMSIG_MOB_EXITED_AREA "mob_exited_area"
```

Comment:
rename to AREA_MOB_FOOBAR for clarity - as is, these would easily be mistaken for signals meant to be sent to mobs, rather than areas